### DISCUSSION TITLE
Post Request Script: Set folder-level environment variable

### DISCUSSION THREAD

**Discussion by bendur** [Kong/insomnia#7736] [Help]
I'm attempting to set a folder-level environment variable based on an API response, but I can't figure out how to set anything other than the currently-selected collection environment.

Is there a way to set folder-level environment variables? 

And/or is there a place where I can see more docs on this?

**Comment by subnetmarco** [comment]
If you click on a folder, do you see this section?

![Screenshot 2024-07-21 at 10 03 31 AM](https://github.com/user-attachments/assets/4364acef-2d39-4c6d-9b60-e8e1f2554b82)

**Comment by subnetmarco** [comment]
@bendur I may have misunderstood your question, if you would like to set folder-level variables from a script, I will let @ihexxa extend on how we could achieve this.

**Comment by bendur** [comment]
Yeah the latter, setting folder-level variables from a script.

**Reply by emaninder** [reply]
similar to options like environment, baseEnvironment and globals, if an option to set a variable at folder level environement will be very helpful.

something like - 
`insomnia.environment.this.set("var", token);`

i expect "this" keyword to be referred as local folder level environment.

**Reply by subnetmarco** [reply]
@ihexxa can you please chime in here?

**Reply by ihexxa** [reply]
Sorry for delayed reply, personally it seems reasonable to expose interfaces for accessing folder-level environment values. We might be able to support this, but before that, we may think about several questions such as 1) how to find the correct target folder to update (as folders are organized as a tree and potentially could share one name) and 2) how to expose an elegant interface to do that.

**Comment by ihexxa** [comment]
Starting a PR for this: #8348

**Reply by ihexxa** [reply]
The PR has been merged and will be released with v11.
